crates/sovereign-sdk/full-node/db/sov-db/src/ledger_db/rpc.rs

```diff
@@ -348,19 +348,22 @@ impl LedgerRpcProvider for LedgerDB {
         &self,
         l2_height: u64,
     ) -> Result<sov_rollup_interface::rpc::SoftConfirmationStatus, anyhow::Error> {
-        let l2_soft_batch = match self.db.get::<SoftBatchByNumber>(&BatchNumber(l2_height)) {
-            Ok(Some(batch)) => batch,
-            _ => {
-                return Err(anyhow::anyhow!(
-                    "Soft confirmation at height {} not processed yet.",
-                    l2_height
-                ))
-            }
-        };
-        let l1_height = l2_soft_batch.da_slot_height;
+        if self
+            .db
+            .get::<SoftBatchByNumber>(&BatchNumber(l2_height))
+            .ok()
+            .flatten()
+            .is_none()
+        {
+            return Err(anyhow::anyhow!(
+                "Soft confirmation at height {} not processed yet.",
+                l2_height
+            ));
+        }
+
         let status = self
             .db
-            .get::<SoftConfirmationStatus>(&SlotNumber(l1_height))?;
+            .get::<SoftConfirmationStatus>(&BatchNumber(l2_height))?;
```

Comment:
instead of calling get twice, a single match is fine, no? return err on None, return value on Some

Reply:
I don't get what you mean? The first get in the if-statement checks the existence of a soft batch while the other one fetches the status

Reply:
sorry, misread.